mobilizon_bots/storage/query.py

```diff
@@ -1,18 +1,68 @@
-from typing import Iterable
+from typing import Iterable, Optional
 
-from mobilizon_bots.event.event import MobilizonEvent, PublicationStatus
+from tortoise.transactions import atomic
+
+from mobilizon_bots.event.event import MobilizonEvent
 from mobilizon_bots.models.event import Event
+from mobilizon_bots.models.publication import Publication, PublicationStatus
+from mobilizon_bots.models.publisher import Publisher
 
 
-async def get_published_events() -> Iterable[MobilizonEvent]:
+async def events_with_status(
+    statuses: list[PublicationStatus],
+) -> Iterable[MobilizonEvent]:
     return map(
         MobilizonEvent.from_model,
-        await Event.filter(
-            publications__status__in=[
-                PublicationStatus.COMPLETED,
-                PublicationStatus.PARTIAL,
-            ]
-        )
+        await Event.filter(publications__status__in=statuses)
+        .prefetch_related("publications")
+        .prefetch_related("publications__publisher")
         .order_by("begin_datetime")
         .distinct(),
     )
+
+
+async def get_published_events() -> Iterable[MobilizonEvent]:
+    return await events_with_status(
+        [
+            PublicationStatus.COMPLETED,
+            PublicationStatus.PARTIAL,
+        ]
+    )
+
+
+async def get_unpublished_events() -> Iterable[MobilizonEvent]:
+    return await events_with_status(
+        [
+            PublicationStatus.WAITING,
+        ]
+    )
+
+
+@atomic("models")
+async def create_unpublished_events(
+    unpublished_events: Iterable[MobilizonEvent], active_publishers: Iterable[str]
+) -> None:
+    # We store only new events, i.e. events whose mobilizon_id wasn't found in the DB.
+    db_unpublished_events_mobilizon_ids = map(
+        lambda event: event.mobilizon_id, await get_unpublished_events()
+    )
+    absent_events = filter(
```

Comment:
I would call these just "unpublished_events"

Reply:
I just removed mobilizon_ids since the query's argument is already called unpublished_events. Should I should change that to `mobilizon_unplublished_events`?